lib/Segments/SegmentsRepository.php

```diff
@@ -169,8 +175,10 @@ public function bulkDelete(array $ids, $type = SegmentEntity::TYPE_DEFAULT) {
   }
 
   public function bulkTrash(array $ids, string $type = SegmentEntity::TYPE_DEFAULT): int {
-    $activelyUsed = $this->newsletterSegmentRepository->getSubjectsOfActivelyUsedEmailsForSegments($ids);
-    $ids = array_diff($ids, array_keys($activelyUsed));
+    $activelyUsedInNewsletters = $this->newsletterSegmentRepository->getSubjectsOfActivelyUsedEmailsForSegments($ids);
+    $activelyUsedInForms = $this->formsRepository->getNamesOfFormsForSegments();
```

Comment:
It seems to me that calling `getNamesOfFormsForSegments()` here means that this method is called twice when deleting a single list and I don't think this is necessary. I understand that we need to call it here when bulk deleting lists. I wonder if it might make sense to split bulkTrash() into two methods. Something like:

```
function bulkTrash(...) {
  // filter out ids that we don't want to delete
  $this->doTrash(...);
}

function doTrash(...) { // maybe we can think of a better name for this method
  // actually delete lists
  // can be called by both bulkTrash() and Segments::trash()
}
```

What do you think?

Reply:
I agree. I was hesitant to change it as this is my first look at the plugin code and I didn't want to cause a regression but it is not optimal.
I am also not sold on the function name `doTrash` but couldn't come up with anything better.

Reply:
Looks good to me. I think we can leave the name `doTrash()` for now as we couldn't come up with a better name.

What I had in mind is call `doTrash()` from inside `bulkTrash()` but the way it is works as well.